### PR TITLE
feat(engine-server): Add support for global HTML attributes and AOM

### PR DESCRIPTION
**Pull request by pmdartus**
## Details

This PR adds support to global HTML attributes and AOM to the `@lwc/engine-server`. In order to do so two new APIs were introduced on the renderer `setProperty` and `getProperty`. Those APIs are invoked when the engine attempts to assign or retrieve a property on a DOM Node.

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

## The PR fulfills these requirements:
* Have tests for the proposed changes been added? ✅ 
* Have you followed [these instructions](../CONTRIBUTING.md#-commit-message-conventions) to clearly describe the issue being fixed or feature enhanced? ✅ 

## GUS work item
W-7552550

### CODE REVIEW

File: packages/@lwc/engine-server/src/renderer.ts [hunk 1]
```diff
@@ -84,6 +90,68 @@ export const renderer: Renderer<HostNode, HostElement> = {
...
+        if (key in node) {
+            return (node as any)[key];
+        }
+
+        if (node.type === HostNodeType.Element) {
+            const attrName = getAttrNameFromPropName(key);
+
+            // Handle all the boolean properties.
+            if (isBooleanAttribute(attrName)) {
+                return this.getAttribute(node, attrName) ?? false;
```
Comment (ravijayaramappa) [inline packages/@lwc/engine-server/src/renderer.ts:103]: How about mirroring the browser behavior of getAttribute(), in fact even the engine-server/renderer's getAttribute() also returns `null`?
```suggestion
                return this.getAttribute(node, attrName) ?? null;
```

File: packages/@lwc/engine-server/src/renderer.ts [hunk 2]
```diff
@@ -84,6 +90,68 @@ export const renderer: Renderer<HostNode, HostElement> = {
...
+
+            // Handle special elements live bindings. The checked property is already handled above
+            // in the boolean case.
+            if (node.name === 'input' && key === 'value') {
+                return this.getAttribute(node, 'value') ?? '';
+            }
+        }
+
+        // eslint-disable-next-line no-console
+        console.warn(`Unexpected "${key}" property access from the renderer`);
```
Comment (ravijayaramappa) [inline packages/@lwc/engine-server/src/renderer.ts:119]:
```suggestion
        if (process.env.NODE_ENV !== 'production') {
            // eslint-disable-next-line no-console
            console.warn(`Unexpected "${key}" property access from the renderer`);
        }
```

File: packages/@lwc/engine-server/src/renderer.ts [hunk 3]
```diff
@@ -84,6 +90,68 @@ export const renderer: Renderer<HostNode, HostElement> = {
...
+            // in the boolean case.
+            if (node.name === 'input' && attrName === 'value') {
+                return isNull(value) || isUndefined(value)
+                    ? this.removeAttribute(node, 'value')
+                    : this.setAttribute(node, 'value', value);
+            }
+        }
+
+        // eslint-disable-next-line no-console
+        console.warn(`Unexpected attempt to set "${key}=${value}" property from the renderer`);
```
Comment (ravijayaramappa) [inline packages/@lwc/engine-server/src/renderer.ts:152]: ditto, wrap the console statement to only warn in dev mode

File: packages/@lwc/engine-server/src/renderer.ts [hunk 4]
```diff
@@ -4,8 +4,14 @@
...
-import { isUndefined, isNull } from '@lwc/shared';
-import { Renderer } from '@lwc/engine-core';
+import {
+    isUndefined,
+    isNull,
+    isBooleanAttribute,
+    isGlobalHtmlAttribute,
+    isAriaAttribute,
+} from '@lwc/shared';
+import { Renderer, getAttrNameFromPropName } from '@lwc/engine-core';
```
Comment (ravijayaramappa) [inline packages/@lwc/engine-server/src/renderer.ts:14]: Would it make sense to move `getAttrNameFromPropName` to @lwc/shared? May be in a separate PR.

File: packages/@lwc/engine-server/src/renderer.ts (hunk 1 above)
Comment (pmdartus) [reply packages/@lwc/engine-server/src/renderer.ts:103]: If the attribute is a boolean attribute, the return value should be `false` get retrieving the property.

**Comment by pmdartus** [comment]
This PR really exhibits the complexity behind the prop vs. attributes allocations made by the template compiler. I think we should attempt to simplify this when reworking the template compiler.
